### PR TITLE
Vulkan 1.3.231

### PR DESCRIPTION
**Pull request by tibequadorian**
Supersedes #39909

#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by JamiKettunen** [comment]
I tested locally and the  [`vkcube` cross-build patch applies fine against `Vulkan-Tools` here](https://github.com/KhronosGroup/Vulkan-Tools/commit/e8234991418c5b0536e342bc849d5f8ed9eb6d6f) with `vkcube` and friend appearing in the package, would you mind including it already? Would be great to have this on non-x86 arches :)

Additionally building of `vkcube` has to be enabled for all arches with the following diff:
```diff
diff --git a/srcpkgs/Vulkan-Tools/template b/srcpkgs/Vulkan-Tools/template
index 4c59112cc1..c2a46ca477 100644
--- a/srcpkgs/Vulkan-Tools/template
+++ b/srcpkgs/Vulkan-Tools/template
@@ -3,9 +3,10 @@ pkgname=Vulkan-Tools
 version=1.3.231.1
 revision=1
 build_style=cmake
-configure_args="-DGLSLANG_INSTALL_DIR=/usr -DBUILD_CUBE=$(vopt_if cube ON OFF)
+build_helper="qemu"
+configure_args="-DGLSLANG_INSTALL_DIR=/usr
  -DVULKAN_HEADERS_INSTALL_DIR=${XBPS_CROSS_BASE}/usr -Wno-dev"
-hostmakedepends="python3 pkg-config $(vopt_if cube glslang)"
+hostmakedepends="python3 pkg-config glslang"
 makedepends="Vulkan-Headers vulkan-loader libxcb-devel libxkbcommon-devel
  wayland-devel wayland-protocols libXrandr-devel"
 short_desc="Official Vulkan tools and utilities"
@@ -14,10 +15,3 @@ license="Apache-2.0"
 homepage="https://www.khronos.org/vulkan/"
 distfiles="https://github.com/KhronosGroup/Vulkan-Tools/archive/sdk-${version}.tar.gz"
 checksum=0388207fdc0986ed8e03092a16d76913c7189d87a0d71eff4a914942b16b552a
-
-build_options="cube"
-desc_option_cube="Build cube vulkan demo"
-
-if [ -z "$CROSS_BUILD" ]; then
-	build_options_default="cube"
-fi

```

QEMU build helper is here to "fix":
```
[1/19] Generating xdg-shell protocol header
FAILED: cube/xdg-shell-client-header.h /builddir/Vulkan-Tools-1.3.231.1/build/cube/xdg-shell-client-header.h 
cd /builddir/Vulkan-Tools-1.3.231.1/build/cube && /usr/aarch64-linux-musl/usr/bin/wayland-scanner client-header /usr/aarch64-linux-musl/usr/share/wayland-protocols/stable/xdg-shell/xdg-shell.xml /builddir/Vulkan-Tools-1.3.231.1/build/cube/xdg-shell-client-header.h
qemu-aarch64-static: Could not open '/lib/ld-musl-aarch64.so.1': No such file or directory
...
```
`wayland-scaller` (from `wayland-devel`, adding it to `hostmakedepends` doesn't change anything) could probably be used from host instead of target right? so patching it and upstreaming the fix shouldn't be too bad (probably), didn't look into it though.

**Comment by tibequadorian** [comment]
Yes thank you, I included it. I think it works better without qemu by using `wayland-scanner` from host.

**Comment by JamiKettunen** [comment]
Ah it was that simple to fix, neat!

**Comment by JamiKettunen** [comment]
At least `vkcube` runs on my `x86_64-glibc` system with an `AMD Radeon RX 580 Series (RADV POLARIS10), type: 2`, will continue running with the updated local packages but I don't think I normally use Vulkan that much

File: srcpkgs/vulkan-loader/template [hunk 1]
```diff
@@ -13,4 +13,10 @@ maintainer="tibequadorian <tibequadorian@posteo.de>"
 license="Apache-2.0"
 homepage="https://www.khronos.org/vulkan/"
 distfiles="https://github.com/KhronosGroup/Vulkan-Loader/archive/sdk-${version}.tar.gz"
-checksum=5dcd42b564804f4c01891ddd429a4a5028113c4cf39b44256689e8f543773134
+checksum=d9d103a6481a19ab9c9b880c1264e3026931741d3184b9930157551453bd4f92
+
+if [ "$XBPS_CROSS_BUILD" ]; then
+	# https://github.com/KhronosGroup/Vulkan-Loader/issues/881
+	# Fixed in master, should be removed in next release
```
Comment (JamiKettunen) [inline srcpkgs/vulkan-loader/template:20]: Couldn't we include https://github.com/KhronosGroup/Vulkan-Loader/commit/d12ff7d48181236b47f808173e044a11423c0f26 here as a patch in the meanwhile before a new sdk tag? I tested locally and it applied just fine with cross-build for `aarch64-glibc` and `aarch64-musl` working :)

Comment (tibequadorian) [reply srcpkgs/vulkan-loader/template]: Sure, thank you for testing!

**Comment by tibequadorian** [comment]
thanks

**Comment by paper42** [comment]
this broke the build of libplacebo

**Comment by tibequadorian** [comment]
I found this https://code.videolan.org/videolan/libplacebo/-/issues/201
The patch seems to work

**Pull request merged**